packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -39,90 +86,199 @@ const ConfirmSignupCode = ({
+        `Email resent. Add ${accountsEmail} to your contacts to ensure a smooth delivery.`,
+        { accountsEmail }
+      );
+      setBanner({
+        visible: true,
+        type: BannerType.success,
+        message: localizedBannerMessage,
+      });
+    } catch (e) {
+      // TODO verify error message copy - This is a new error message
+      const localizedBannerMessage = ftlMsgResolver.getMsg(
+        'confirm-signup-code-error-message',
+        'Something went wrong. A new code could not be sent.'
       );
-      setCodeErrorMessage(codeRequiredError);
+      setBanner({
+        visible: true,
+        type: BannerType.error,
+        message: localizedBannerMessage,
+      });
     }
+  }
+
+  function alertSuccessAndGoForward() {
+    alertBar.success(
+      ftlMsgResolver.getMsg(
+        'confirm-signup-code-success-alert',
+        'Account confirmed successfully'
+      )
+    );
+    // TODO redirect elsewhere if relying party
+    navigate('/settings', { replace: true });
+  }
+
+  async function verifySession(code: string) {
+    logViewEvent(`flow.${viewName}`, 'submit', REACT_ENTRYPOINT);
     try {
-      // Check confirmation code
-      // Log success event (verification.success)
-      // Display success message
-      // Handle newsletter subscription
+      await account.verifySession(code);
+
+      // FOLLOW-UP: does not yet exist in Settings
+      // BEFORE: notifier.trigger('verification.success');
+      logViewEvent(
+        `flow.${viewName}`,
+        'verification.success',
+        REACT_ENTRYPOINT
+      );
+      // FOLLOW-UP: enable once state can be accessed from previous component
+      // FOLLOW-UP: notifier does not exist in Settings
+      // BEFORE: this.notifier.trigger('flow.event', {event: 'newsletter.subscribed',})
+      // if (newsletters) {
+      //   logViewEvent(`flow.${viewName}`, 'newsletter.subscribed', REACT_ENTRYPOINT);
+      // }
+
+      // FOLLOW-UP: Broker not yet implemented
+      // await broker.invokeBrokerMethod('afterSignUpConfirmationPoll', account);
+      alertSuccessAndGoForward();
     } catch (e) {
-      // TODO: error handling, error message confirmation
-      // Possible AuthErrors to display in tooltip:
-      //   - 'INVALID_EXPIRED_SIGNUP_CODE' = 'Invalid or expired confirmation code'
-      //   - 'OTP_CODE_REQUIRED' = 'Please enter confirmation code'
-      //   - 'INVALID_OTP_CODE' = 'Valid code required'
-      // all other errors: alertBar.error(errorConfirmSignupCode);
+      // TODO log error
+      const localizedErrorMessage = ftlMsgResolver.getMsg(
+        composeAuthUiErrorTranslationId(e),
+        e.message
+      );
+      // If the error is one of the three indicated types, display error in input tooltip
+      if (
+        e.errno ===
+        (AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno ||
+          AuthUiErrors.OTP_CODE_REQUIRED.errno ||
+          AuthUiErrors.INVALID_OTP_CODE.errno)
+      ) {
```

Comment:
I don't think this is right. IIUC, this will take the first truthy value from the `||` statements in the parens, not do 3 separate lookups; so it will probably always pass on `e.errno === AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno` condition.

Possibly either something like three separate conditional checks:
```js
if (
  e.errno === AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno ||
  e.errno === AuthUiErrors.OTP_CODE_REQUIRED.errno ||
  e.errno === AuthUiErrors.INVALID_OTP_CODE.errno
) {
  // Yep
} else {
  // Nope
}
```

Or, using `Array#includes()`:
```js
if ([
  AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno,
  AuthUiErrors.OTP_CODE_REQUIRED.errno,
  AuthUiErrors.INVALID_OTP_CODE.errno
].includes(e.errno)) {
  // Yep
} else {
  // Nope
}
```

Reply:
Good catch Peter; Valerie, I'd probs go with Peter's first suggestion here, but includes is fine too.

Reply:
Alternatively, the [`in` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in):
```js
if (e.errno in [
  AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno,
  AuthUiErrors.OTP_CODE_REQUIRED.errno,
  AuthUiErrors.INVALID_OTP_CODE.errno
]) {
  console.log("Yep");
} else {
  console.log("Nope");
}
```

Or even [`Array#some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some), if you're feeling whimsical:
```js
if ([
  AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE,
  AuthUiErrors.OTP_CODE_REQUIRED,
  AuthUiErrors.INVALID_OTP_CODE
].some(({ errno }) => errno === e.errno)) {
  console.log("Yep");
} else {
  console.log("Nope");
}
```

#WhimsyBasedDevelopment